src/test/java/fr/raluy/chocoratage/KeysBufferTest.java

```diff
@@ -30,15 +33,52 @@ public void shouldAddStringToEmptyBuffer() {
     @Test
     public void shouldAddStringToBuffer() {
         addStringToBuffer("TEST ");
-        addStringToBuffer("HAVE");
+        addStringToBuffer("HAS\t\n");
+        addStringToBuffer("FOUR\f ");
+        addStringToBuffer("WORDS");
 
-        assertThat(keysBuffer.toString()).isEqualTo("test have");
+        assertThat(keysBuffer.toString()).isEqualTo("test has four words");
     }
 
     @Test
-    public void shouldNotAddNull() {
-        addStringToBuffer("TEST");
-        keysBuffer.add(null);
+    public void shouldAddAccentedCharacters() {
+        addStringToBuffer("TEST ÉùàçØöğ");
+        assertThat(keysBuffer.toString()).isEqualTo("test éùàçøöğ");
+    }
+
+    @Test
+    public void shouldAddNonLatinStringToBuffer() {
+        addStringToBuffer("La"); // Latin
+        addStringToBuffer("αΨ"); // Greek
+        addStringToBuffer("Жц"); // Cyrillic
+        addStringToBuffer("թխ"); // Armenian
+        addStringToBuffer("جؼ"); // Arabic
+        addStringToBuffer("שמ"); // Hebrew
+        addStringToBuffer("葵袷"); // Kanji
+        addStringToBuffer("ばふ"); // Hiragana
+        addStringToBuffer("하함"); // Korean
+        addStringToBuffer("腌盒"); // Chinese
+        addStringToBuffer("༺༪"); // Tibetan
+        addStringToBuffer("บพ"); // Thai
+        addStringToBuffer("ങജ"); // Malayalam
+        addStringToBuffer("ছয়"); // Bengali
+        addStringToBuffer("ఞఌ"); // Telugu
+        assertThat(keysBuffer.toString()).isEqualTo("laαψжцթխجؼשמ葵袷ばふ하함腌盒บพങജছয়ఞఌ");
+    }
+
+    @Test
+    public void shouldNotAddNonLetterOrDigit() {
+        addStringToBuffer("♥♥♥");
```

Comment:
Shouldnt we care about unicode chars depicting chocolatines?

Reply:
Yep, but imo now we can blacklist this char. I'll do a UT for this :)

Reply:
see #14